### ISSUE TITLE
Make use of libvirt's volume API

### ISSUE THREAD

**Issue by zeenix** [code-ready/machine-driver-libvirt#39]
We should be doing this, if not for anything else then[ at least to avoid copying images without having to rely on external `qemu-img` command](https://github.com/code-ready/machine-driver-libvirt/pull/38#discussion_r353086695).

**Comment by zeenix** [comment]
Just remembered that using volume API also implies storage pool API AFAIK. This isn't a problem on it's own. We can create a `dir` pool and put the volume there but it means that the location of disk image will be different on different platforms then and a bit more work than I previously thought. The latter is not a problem if the former is not though. WDYT?

**Comment by cfergeau** [comment]
> but it means that the location of disk image will be different on different platforms

I don't understand this one. libvirt is only used on linux.

**Comment by zeenix** [comment]
> I don't understand this one. libvirt is only used on linux.

that's why the path will be different on Linux (sorry could have worded it better) and that's why on Linux the path will be different. Not sure if it's an issue though.

**Comment by cfergeau** [comment]
Still not quite clear what you mean :( The `dir` pool could point at `~/.crc/machines/crc`, so the image would still be in the same location as other platforms. I'm totally missing your point and concern :(

**Comment by zeenix** [comment]
> The `dir` pool could point at `~/.crc/machines/crc`

Yeah but then the dir belongs to the pool and deletion will remove it, which may not always be what we want.

**Comment by cfergeau** [comment]
Ok, so you are suggesting pointing the pool at a path outside of `~/.crc/machines/crc` ?
Removing the pool behind crc back sounds like something unsupported, so I would say losing `~/.crc/machines/crc` in that case would not be that bad.

**Comment by zeenix** [comment]
> Ok, so you are suggesting pointing the pool at a path outside of `~/.crc/machines/crc` ?

I was thinking of a subdir there actually but now I also remembered that currently the driver keeps room for multiple machines in the future (`crc` machine name is not hard coded) so a directory outside that would be best, yes.

> Removing the pool behind crc back sounds like something unsupported,

No, I wasn't thinking of that. I was thinking that we might not want everything in that dir to be always deleted at the same time as we delete the pool, in the future. Also the other way around: not deleting the pool but deleting the dir still. The main issue i see here is the dir becoming a property of the pool and it's not exactly the case here.
